Snippets/ASP/ASTP_3/Behaviors.cs

```diff
@@ -84,34 +84,39 @@ public override async Task Invoke(IIncomingLogicalMessageContext context, Func<T
 
 class OrderIdAsPartitionKeyBehavior : Behavior<IIncomingLogicalMessageContext>
 {
-    public OrderIdAsPartitionKeyBehavior(IProvidePartitionKeyFromSagaId partitionKeyFromSagaId)
-    {
+    private static readonly ILog Log = LogManager.GetLogger<OrderIdAsPartitionKeyBehavior>();
+    
+    private readonly IProvidePartitionKeyFromSagaId partitionKeyFromSagaId;
+
+    public OrderIdAsPartitionKeyBehavior(IProvidePartitionKeyFromSagaId partitionKeyFromSagaId) =>
         this.partitionKeyFromSagaId = partitionKeyFromSagaId;
-    }
 
     public override async Task Invoke(IIncomingLogicalMessageContext context, Func<Task> next)
     {
         var correlationProperty = SagaCorrelationProperty.None;
+
         if (context.Message.Instance is IProvideOrderId provideOrderId)
         {
-            var partitionKeyValue = provideOrderId.OrderId;
-            correlationProperty = new SagaCorrelationProperty("OrderId", partitionKeyValue);
+            Log.Debug($"Order ID: '{provideOrderId.OrderId}'");
+
+            correlationProperty = new SagaCorrelationProperty("OrderId", provideOrderId.OrderId);
```

Comment:
Honestly, I liked the partitionKeyValue as it underlines that it's used as a partition key.

Reply:
Discussed with @danielmarbach and we decided to leave as-is. One guiding factor is that parameter name in `SetPartitionKey` is `correlationProperty`, and it was not always be used as the partition key, so this call would look misleading:

```c#
partitionKeyFromSagaId.SetPartitionKey<OrderSagaData>(context, partitionKeyValue )
```